FormWidget.js

```diff
@@ -145,14 +167,26 @@ define([
 					// disabled property.  Note that on IE every element has a disabled property, so it's hard to
 					// test if it's real or not.
 					node.disabled = this.disabled;
+					node.required = this.required;
 
-					// Set aria-disabled attribute if necessary, but try to avoid setting it redundantly.
+					// Set aria-disabled and required but try to avoid setting it redundantly.
 					if (!(/^(button|input|select|textarea|optgroup|option|fieldset)$/i.test(node.tagName))) {
 						node.setAttribute("aria-disabled", "" + this.disabled);
+						node.setAttribute("aria-required", this.required);
+					}
+					
+					//set the focus nodes input id
+					if ((/^(input)$/i.test(node.tagName))) {
+						if (inputId && inputIdUnset) {
+							inputIdUnset = false;
+							node.id = inputId;
+						}
```

Comment:
I suppose there could be other node types too, like `<textarea>`.   (I was going to say `<button>` but then I remembered that buttons don't take external labels.).

Also, if the field already has an existing ID then you probably shouldn't override it?